spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/source/BatchDataReader.java

```diff
@@ -50,15 +50,17 @@ class BatchDataReader extends BaseBatchReader<FileScanTask>
   BatchDataReader(
       SparkInputPartition partition,
       ParquetBatchReadConf parquetBatchReadConf,
-      OrcBatchReadConf orcBatchReadConf) {
+      OrcBatchReadConf orcBatchReadConf,
```

Comment:
For this and parquet we should probably add to the Conf object. Ie OrcBatchReadConf.cacheDeleteFiles

Reply:
I had considered that, but it became harder to pass the flag back to parent class (for SparkDeleteFilter creation) since BatchDataReader is common class for both parquet and orc (I would have to do null checks and get the flag) 

Also, this property is a format agnostic one, while the `<format>BatchReadConf` are for format specific properties.

Reply:
Sorry for the delay, poking around a bit I think if we put delete file can cache information in the SparkInputPartition we can probably plumb this all the way through without changing as many constructors.

Reply:
No problem!

Thanks for the idea. Now I followed what we did for `caseSensitive()`. It did simplify the changes a bit.